gammapy/utils/interpolation.py

```diff
@@ -253,13 +253,17 @@ def interpolate_profile(x, y, interp_scale="sqrt"):
         of parabolic shape, a "sqrt" scaling is recommended. In other cases or
         for fine sampled profiles a "lin" can also be used.
         Default is "sqrt".
+    method : {“linear”, “nearest”, “slinear”, “cubic”, “quintic”, “pchip”}
+        The method of interpolation to perform.
+        Default is “pchip” (better for profile of parabolic shape).
+    **kwargs : dict
+        Keyword arguments passed to `RegularGridInterpolator`.
 
     Returns
     -------
     interp : `ScaledRegularGridInterpolator`
         Interpolator.
     """
-    sign = np.sign(np.gradient(y))
```

Comment:
Why would we need this line ?

Reply:
The `interpolate_profile` does a square root scaling of the values. The profile is assumed to be a parabola and one gets the `abs` function if one does not correct for the sign. With the `abs` function you cannot interpolate linearly across zero. I think @AtreyeeS interpretation is correct, that the `interpolate_profile` assume the profiles goes through zero. So shifting the profile is probably the correct fix. Or just change to scipy quadratic interpolation, as in @AtreyeeS example.

Reply:
This is because you have a sampling point at, or very close to the minimum. The sign handling allows to interpolate correctly between samples, even if the minimum is between them. See:

```python3
import numpy as np
import matplotlib.pyplot as plt
from gammapy.utils.interpolation import interpolate_profile, ScaledRegularGridInterpolator

x = np.linspace(-5, 5, 6)
y = x ** 2

x_fine = np.linspace(-5, 5, 100)

interp = interpolate_profile(x, y)
interp_2 = ScaledRegularGridInterpolator(
        points=(x,), values=y, values_scale="sqrt"
    )

plt.plot(x_fine, interp((x_fine,)), color="tab:orange")
plt.plot(x_fine, interp_2((x_fine,)), color="tab:green")
plt.scatter(x, y)
```
Which gives:

<img width="544" alt="image" src="https://github.com/gammapy/gammapy/assets/3715928/3042ce69-90e8-4e44-b7ca-5061e6188a9d">

Reply:
Using cubic spline with lin scale seems ok for that case :
![image](https://github.com/gammapy/gammapy/assets/25845654/f53df927-da9e-4e8d-9c56-d81859517367)